exercises/xorcism/tests/xorcism.rs

```diff
@@ -0,0 +1,303 @@
+use rstest::rstest;
+use rstest_reuse::{self, *};
+#[cfg(feature = "io")]
+use std::io::{Read, Write};
+use xorcism::Xorcism;
+use hexlit::hex;
+
+#[test]
+fn identity() {
+    let mut xs = Xorcism::new(&[0]);
+    let data = "This is super-secret, cutting edge encryption, folks.";
+
+    assert_eq!(
+        xs.munge(data.as_bytes()).collect::<Vec<_>>(),
+        data.as_bytes()
+    );
+}
+
+#[test]
+#[ignore]
+fn munge_output_has_len() {
+    let mut xs = Xorcism::new(&[0]);
+    let data = "The output must have a definite length";
+    assert_eq!(xs.munge(data.as_bytes()).len(), 38);
+}
+
+#[test]
+#[ignore]
+fn statefulness() {
+    // we expect Xorcism to be stateful: at the end of a munging run, the key has rotated.
+    // this means that until the key has completely rotated around, equal inputs will produce
+    // unequal outputs.
+    let key = &[0, 1, 2, 3, 4, 5, 6, 7];
+    let input = &[0b1010_101, 0b0101_0101];
```

Comment:
Although these two elements have the same value, they are formatted differently and thus may look confusing. Notice how the first one has three elements after the underscore, and the second has four. I don't suppose this is intentional?

Reply:
Good catch! That was an accident. ebfa0e2